### DISCUSSION TITLE
Add support for rendering d2 diagrams

### DISCUSSION THREAD

**Discussion by prologic** [outline/outline#5925] [Ideas]
Similar to the feature recently relased:

> Mermaid diagram rendering performance improved by @antran22 in https://github.com/outline/outline/pull/5852

It would be great if we could add support for rendering [d2](https://d2lang.com/) diagrams 👌

**Comment by antran22** [comment]
This seems pretty hard to manage, since D2 is a Go library and run on the server side.
Might be interesting to wait until https://github.com/terrastruct/d2/pull/436 is complete, we can integrate D2 as a fully browser side library.

**Reply by prologic** [reply]
Sounds good 👌

**Reply by fruworg** [reply]
> merged commit [5838760](https://github.com/terrastruct/d2/commit/5838760d199d0a36327a8d65f1cbb52d47fbe9ed) into terrastruct:master [on Dec 30, 2024](https://github.com/terrastruct/d2/pull/436#event-15785583224)

hi! any updates?
